src/source/mod.rs

```diff
@@ -168,6 +168,53 @@ impl LoaderChecksum {
   }
 }
 
+impl fmt::Display for LoaderChecksum {
+  fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+    write!(f, "{}", self.0)
+  }
+}
+
+pub trait Locker {
+  fn get_checksum(&self, specifier: &ModuleSpecifier)
+    -> Option<LoaderChecksum>;
+  fn has_checksum(&self, specifier: &ModuleSpecifier) -> bool;
+  fn set_checksum(
+    &mut self,
+    specifier: &ModuleSpecifier,
+    checksum: LoaderChecksum,
+  );
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct HashMapLocker(HashMap<ModuleSpecifier, LoaderChecksum>);
```

Comment:
Where would we use this one? In `deno vendor`?

Reply:
Just in the tests probably.